src/templates_front/templator_front.org

```diff
@@ -521,19 +532,23 @@ def string_of_error(return_code: int) -> str:
    #+begin_src c :tangle prefix_front.h
 typedef int32_t back_end_t;
 
-#define TREXIO_HDF5             ( (back_end_t) 0 )
-#define TREXIO_TEXT             ( (back_end_t) 1 )
-/*#define TREXIO_JSON             ( (back_end_t) 2 )*/
+#define TREXIO_TEXT             ( (back_end_t) 0 )
+#ifdef HAVE_HDF5
+#define TREXIO_HDF5             ( (back_end_t) 1 )
 #define TREXIO_INVALID_BACK_END ( (back_end_t) 2 )
+#else
+#define TREXIO_INVALID_BACK_END ( (back_end_t) 1 )
+#endif
+/*#define TREXIO_JSON             ( (back_end_t) 2 )*/
 
 #define TREXIO_DELIM            "\n"
    #+end_src
 
 *** Fortran
 
   #+begin_src f90 :tangle prefix_fortran.f90
-  integer(trexio_backend), parameter :: TREXIO_HDF5 = 0
-  integer(trexio_backend), parameter :: TREXIO_TEXT = 1
+  integer(trexio_backend), parameter :: TREXIO_TEXT = 0
+  integer(trexio_backend), parameter :: TREXIO_HDF5 = 1
 ! integer(trexio_backend), parameter :: TREXIO_JSON = 2
   integer(trexio_backend), parameter :: TREXIO_INVALID_BACK_END = 2
```

Comment:
Here the Fortran is inconsistent with the C for the invalid back-end. The simplest would be to keep the TREXIO_HDF5=1 even if HDF5 is not present.